### PR TITLE
Kills a bunch of harddels as well as GC failures. Big PR.

### PR DESCRIPTION
**Pull request by nikothedude**
## About The Pull Request
Changes so many things. I'll give brief descriptions.

Fixes the assembly GC failure when a assembly is qdelled. Fixes the qdel that happens when a few mobs with reagent containers qdel. I THINK it fixes the constant reagent/metabolism GC failure. Converts mob targetting vars to weakref to prevent GC failure. A lot more. I've tested it, a bit, this shouldn't break anything. 

Removed the atom/movable/Del() override because it might've been causing lag. If you want to diagnose this kinda stuff, do it locally. Don't keep your debug stuff on live.

Also does more that I forgot about.

THIS DOES NOT FIX THE PIPE HARDDELS. I have no fucking idea why this happens IT JUST DOES.

	
</details>

## Changelog
:cl:
fix: Fixes TONS of GC failures
refactor: Switches mob targetting vars to weakref
:cl\:

### CODE REVIEW

**Comment by nikothedude** [comment]
Will get rid of the GC failures brought on by https://github.com/sojourn-13/sojourn-station/pull/3134. 

Also branched off of it.

**Comment by Trilbyspaceclone** [comment]
Linters are failing

File: code/_compile_options.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-//#define TESTING //By using the testing("message") proc you can create debug-feedback for people with this
+#define TESTING //By using the testing("message") proc you can create debug-feedback for people with this
```
Comment (r4d6) [inline code/_compile_options.dm:1]: Don't forget to comment out the testing stuff.

**Comment by r4d6** [comment]
> Linters are failing

It's because of a coded warning whenever the server start in Testing Mode.

**Comment by nikothedude** [comment]
Yeah, I know about the warnings, the testing stuff, etc. The reason it's a draft is because all this stuff remains.

File: code/_compile_options.dm (hunk 1 above)
Comment (nikothedude) [reply code/_compile_options.dm:1]: Noted

File: code/_compile_options.dm [hunk 2]
```diff
@@ -12,7 +12,7 @@
 #define DATUMVAR_DEBUGGING_MODE
 
 ///Used to find the sources of harddels, quite laggy, don't be surpised if it freezes your client for a good while
-//#define REFERENCE_TRACKING
+#define REFERENCE_TRACKING
```
Comment (hex37) [inline code/_compile_options.dm:15]: Should we turn this back so it doesn't lag the server?

Comment (nikothedude) [reply code/_compile_options.dm]: Yeah, absolutely, which is why its a draft and all

**Comment by nikothedude** [comment]
Note this hasn't been THROUGHOUGLY tested although I fail to see how this can cause any problems.

**Pull request merged**